phenol-annotations/src/main/java/org/monarchinitiative/phenol/annotations/assoc/OrphaGeneToDiseaseParser.java

```diff
@@ -37,77 +37,28 @@
   private static final String REFERENCE = "Reference";
   private static final String SOURCE = "Source";
 
-  private boolean inDisorder = false;
-  private boolean inGeneList = false;
-  private boolean inGene = true;
-  private boolean inDisorderGeneAssociation = false;
-  private boolean inDisorderGeneAssociationList = false;
-  private boolean inExternalReference = false;
-  private boolean inOMIMReference = false;
 
   /**
-   * Key: e.g., ORPHA:163746 (for Peripheral demyelinating neuropathy-central dysmyelinating
-   * leukodystrophy-Waardenburg syndrome-Hirschsprung disease). Value, corresponding gene symbol(s),
-   * e.g., SOX10
+   * @return map with key: e.g., ORPHA:163746 (for Peripheral demyelinating neuropathy-central dysmyelinating
+   * leukodystrophy-Waardenburg syndrome-Hirschsprung disease).
+   * value: corresponding gene symbol(s), e.g., SOX10
    */
-  private final Multimap<TermId, Gene> orphaDiseaseToGeneMultiMap;
```

Comment:
I don't understand the fanaticism for removing Guava with this PR. Sure, some things especially ImmutableList should use the standard library, but Multimap has a fairy valid reason to exist.

Is this a push to try and move to a fully JPMS library? In which case, fair enough.

Reply:
Yes, the idea is that the dependencies that prevents us using `jlink` should go, and guava is one of them.

Their view on moving from having `Automatic-Module-Name` in `MANIFEST.MF` to a fully modular JAR is discussed [here](https://github.com/google/guava/issues/2970), and I do not see any signs of moving forward to modular JAR in not so distant future.

`Multimap` looks useful, but from what I read in Guava's wiki, it's mostly a convenience. The cost of removing `Multimap` is not too high if we provide several container classes, such as `HpoDisease`, instead of providing `Multimap<TermId, TermId> diseaseToGeneMap` or `Multimap<TermId, TermId> geneToDiseaseMap`. I hope that the intended usage of the containers will be self-explanatory.

Guava is the biggest step in making `phenol` a modular library.

Reply:
I think there are many places where we use Phenol as a library, and they all will benefit from modularization. For instance _HpoCaseAnnotator_ and _SvAnna_  can be made into standalone tools using `jlink`, to increase the _SWAG_ factor, and to simplify long-term maintenance.

I went the whole hog, and added `Sets.intersection` & `Sets.union` methods, so now Guava is really gone. This had some surprising consequences - `phenol-io` tests failed and I had to tweak _Surefire_ configuration to let `com.fasterxml.jackson.datatype.guava` read unnamed modules, in order to make the tests pass.

Even though Phenol is not 100% modular yet, I think we are as nice as we can be.

- `phenol-core` depends on modular `jgraph` that has non-modular transitive dependency (`jheaps`), meaning that we can become modular if we write our own graphs, or use modular graph library (not super hard).
- `phenol-io` still uses non-modular libraries for loading JSON, however these are libraries that are under our control. 
- `phenol-annotations` is modular
- `phenol-analysis` becomes modular as soon as we re-implement or drop `commons-codec` functionality in `ObjHexStringConverter` (not super hard)
- `phenol-cli` depends on `phenol-io`, but otherwise is OK

Reply:
There is no shame in copy-pasting a class or two from an other open source library, providing it is done so with full attribution and licence, for the sake of removing all the unused functionality.

I'll need to revisit `obographs-core` and bring it into the modular world, if possible. Shouldn't be too hard as it only requires Jackson. The OLW-API part might  be more of a challenge.

Can you document these on an issue so we can revisit them from time to time. Dropping JGraphT, for example might be a bit of a pain.